packages/flutter/test/material/theme_test.dart

```diff
@@ -367,6 +367,53 @@ void main() {
 
     expect(actualFontSize, _kMagicFontSize);
   });
+
+  testWidgets('Default Theme provides all basic TextStyle properties', (WidgetTester tester) async {
+    ThemeData theme;
+    await tester.pumpWidget(new Directionality(
+      textDirection: TextDirection.ltr,
+      child: new Builder(
+        builder: (BuildContext context) {
+          theme = Theme.of(context);
+          return const Text('A');
+        },
+      ),
+    ));
+
+    List<TextStyle> extractStyles(TextTheme textTheme) {
+      return <TextStyle>[
+        textTheme.display4,
+        textTheme.display3,
+        textTheme.display2,
+        textTheme.display1,
+        textTheme.headline,
+        textTheme.title,
+        textTheme.subhead,
+        textTheme.body2,
+        textTheme.body1,
+        textTheme.caption,
+        textTheme.button,
+      ];
+    }
+
+    for (TextTheme textTheme in <TextTheme>[theme.textTheme, theme.primaryTextTheme, theme.accentTextTheme]) {
+      for (TextStyle style in extractStyles(textTheme)) {
+        expect(style.color, isNotNull);
+        expect(style.fontFamily, isNotNull);
+        expect(style.package, null);
+        expect(style.fontSize, isNotNull);
+        expect(style.fontWeight, isNotNull);
+        expect(style.fontStyle, null);
+        expect(style.letterSpacing, null);
+        expect(style.wordSpacing, null);
+        expect(style.textBaseline, isNotNull);
+        expect(style.height, null);
+        expect(style.decoration, TextDecoration.none);
+        expect(style.decorationColor, null);
+        expect(style.decorationStyle, null);
+      }
```

Comment:
It's too bad there's no way to force this test to be updated if any properties are added to `TextStyle`...

Reply:
There is one way. Have a class in this file that "implements" TextStyle. The analyzer will flag this file whenever we add a member.

Reply:
Wow, this trick is so devious that I implemented it.